### DISCUSSION TITLE
How to handle reactive with spreading

### DISCUSSION THREAD

**Discussion by phungleson** [solidjs/solid#1016] [Q&A]
I have updated to recent version of Solidjs an reactive behaviour seems to change.

[Repo code](https://playground.solidjs.com/?hash=-796555442&version=1.4.1)

Basically it looks like this.

```
const defaults = (props: Partial<Props>): Props => {
  console.log("abc");

  return {
    // some defaults
    ...props,
  }
}

// then in test, something like this

render(() => <Component {...defaults(props)} />, document.body);
```

The issue is that it prints out `abc` every time I access `props.someAttr` inside `Component`.

`lxsmnsyc` suggested that it might be related to reactive spreading.

I would like to best handle this case, which is removal of redundant reactivity with least amount of code change?

**Comment by lxsmnsyc** [comment]
Use `mergeProps`
